cmd/eskip/media.go

```diff
@@ -93,25 +103,36 @@ func validateSelectWrite(cmd command, media []*medium) (input, output *medium, e
 		return nil, nil, missingInput
 	}
 
-	if out == nil {
-		var err error
-		out, err = processEtcdArgs(defaultEtcdUrls, defaultEtcdPrefix)
-		if err != nil {
-			return nil, nil, err
+	return in, out, nil
+}
+
+func validateSelectDelete(media []*medium) (in, out *medium, err error) {
+	if len(media) == 0 {
+		return nil, nil, nil
+	}
+
+	if len(media) > 2 {
+		return nil, nil, tooManyInputs
+	}
+
+	for _, m := range media {
+
+		if m.typ == etcd || m.typ == innkeeper {
+			out = m
+		} else {
+			in = m
 		}
 	}
 
+	if in == nil {
+		return nil, nil, missingInput
+	}
+
 	return in, out, nil
 }
 
-// Validate media from args for the current command, and select input and/or output.
+// Validates media from args for the current command, and selects input and/or output.
 func validateSelectMedia(cmd command, media []*medium) (input, output *medium, err error) {
-	switch cmd {
-	case check, print:
-		return validateSelectRead(media)
-	case upsert, reset, delete:
-		return validateSelectWrite(cmd, media)
-	default:
-		return nil, nil, invalidCommand
-	}
+	// cmd should be present and valid
+	return commandToValidations[cmd](media)
```

Comment:
we already know where each command (print, upsert) goes, so we don't need to decide this every time we call it. The advantage of having separate functions is also that we give name to concepts. So the readability is improvoed, as we split the commands into Read and Write commands.